### PR TITLE
[WIP] Contract archive integration

### PR DESCRIPTION
**Pull request by morty**

### CODE REVIEW

File: ckanext/dgu/controllers/data.py [hunk 1]
```diff
@@ -193,11 +193,28 @@ def openspending_publisher_report(self, id):
...
+
     def viz_front_page(self):
         return render('viz/front_page.html')
 
+    def contracts_archive(self, relative_url):
+        import requests, urlparse
+        from pylons import response
+
+        headers = {'X-Script-Name': '/data/contracts-archive'}
+        url = urlparse.urljoin('http://46.43.41.30/', relative_url)
```
Comment (davidread) [inline ckanext/dgu/controllers/data.py]: TODO: make this a config option

File: ckanext/dgu/controllers/data.py [hunk 2]
```diff
@@ -193,11 +193,28 @@ def openspending_publisher_report(self, id):
...
+        r = requests.get(url, headers=headers, stream=True)
+
+        if relative_url.startswith('/static/'):
+            # CSS will only get loaded if it has the right content type
+            response.content_type = r.headers.get('content-type', 'text/html')
+            return r.raw.read() # Some of the static files are binary
+        else:
+            c.content = r.text
```
Comment (davidread) [inline ckanext/dgu/controllers/data.py]: pass this in to render() as a param rather than using c.

**Comment by davidread** [comment]
Looks good

**Comment by davidread** [comment]
I thought this was merged?

**Comment by morty** [comment]
I think I split this up into two PRs in the end.

**Pull request closed without merging**